File: iot/app/src/main/java/siarhei/luskanau/example/iot/IotActivity.java
package siarhei.luskanau.example.iot;

import android.os.Build;
import android.os.Bundle;

import com.google.android.things.pio.PeripheralManager;
import com.google.gson.Gson;

import timber.log.Timber;

public class IotActivity extends CameraActivity {

    private final Gson GSON = new Gson();

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Timber.i("Starting IotActivity");
    }

    @Override
    protected void onResume() {
        super.onResume();
        buildInfo();
        pioInfo();
    }

    private void buildInfo() {
        Timber.i("Build.DEVICE: %s", GSON.toJson(Build.DEVICE));
        Timber.i("Build.BOARD: %s", GSON.toJson(Build.BOARD));
        Timber.i("Build.HARDWARE: %s", GSON.toJson(Build.HARDWARE));
        Timber.i("Build.MODEL: %s", GSON.toJson(Build.MODEL));
        Timber.i("Build.PRODUCT: %s", GSON.toJson(Build.PRODUCT));
        Timber.i("Build.DISPLAY: %s", GSON.toJson(Build.DISPLAY));
        Timber.i("Build.ID: %s", GSON.toJson(Build.ID));
    }

    private void pioInfo() {
        try {
            final PeripheralManager peripheralManager = PeripheralManager.getInstance();
            Timber.i("GpioList: %s", GSON.toJson(peripheralManager.getGpioList()));
            Timber.i("I2cBusList: %s", GSON.toJson(peripheralManager.getI2cBusList()));
            Timber.i("PwmList: %s", GSON.toJson(peripheralManager.getPwmList()));
            Timber.i("SpiBusList: %s", GSON.toJson(peripheralManager.getSpiBusList()));
            Timber.i("UartDeviceList: %s", GSON.toJson(peripheralManager.getUartDeviceList()));
        } catch (final Throwable e) {
            Timber.e(e);
        }
    }
}

File: iot/app/src/main/java/siarhei/luskanau/example/iot/IotActivity.kt
package siarhei.luskanau.example.iot

import android.os.Build
import android.os.Bundle
import com.google.android.things.pio.PeripheralManager
import com.google.gson.Gson
import timber.log.Timber

class IotActivity : CameraActivity() {

    private val gson = Gson()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Timber.d("Starting IotActivity")
    }

    override fun onResume() {
        super.onResume()
        buildInfo()
        pioInfo()
    }

    private fun buildInfo() {
        Timber.d("Build.DEVICE: %s", gson.toJson(Build.DEVICE))
        Timber.d("Build.BOARD: %s", gson.toJson(Build.BOARD))
        Timber.d("Build.HARDWARE: %s", gson.toJson(Build.HARDWARE))
        Timber.d("Build.MODEL: %s", gson.toJson(Build.MODEL))
        Timber.d("Build.PRODUCT: %s", gson.toJson(Build.PRODUCT))
        Timber.d("Build.DISPLAY: %s", gson.toJson(Build.DISPLAY))
        Timber.d("Build.ID: %s", gson.toJson(Build.ID))
    }

    private fun pioInfo() =
            try {
                val peripheralManager = PeripheralManager.getInstance()
                Timber.d("GpioList: %s", gson.toJson(peripheralManager.gpioList))
                Timber.d("I2cBusList: %s", gson.toJson(peripheralManager.i2cBusList))
                Timber.d("PwmList: %s", gson.toJson(peripheralManager.pwmList))
                Timber.d("SpiBusList: %s", gson.toJson(peripheralManager.spiBusList))
                Timber.d("UartDeviceList: %s", gson.toJson(peripheralManager.uartDeviceList))
            } catch (e: Throwable) {
                Timber.e(e)
            }
}
